### PR TITLE
move to goal line fsm

### LINKED ISSUE #2078: move to goal line fsm
### Acceptance criteria

goalie moves to the goal line and faces the enemy

### PR DESCRIPTION
**Pull request by xyztiger**
### Description

Added move to goal line fsm for penalty kick goalie.

### Testing Done

fsm tests, simulated tactic tests

### Resolved Issues

resolves #2078 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/tactic/tactic_visitor.h [hunk 1]
```diff
@@ -25,6 +25,8 @@ class ShootGoalTactic;
 class StopTactic;
 class StopTestTactic;
 class DribbleTactic;
+class MoveToGoalLineTactic;
+class MoveToGoalLineTestTactic;
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/tactic_visitor.h]: left over
```suggestion
```

File: src/software/estop/uart_communication.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
+
```
Comment (jonathanlew) [inline src/software/estop/uart_communication.cpp]: remove this file

**Review by jonathanlew — COMMENTED** [review]
just got a few comments

**Review by jonathanlew — APPROVED** [review]
LGTM! :+1:

**Review by akhilveeraghanta — APPROVED** [review]
💯

**Pull request merged**